lib/Extractor/Solver.cpp

```diff
@@ -316,8 +320,18 @@ class MemCachingSolver : public Solver {
                           std::vector<std::pair<Inst *, llvm::APInt>> *Model)
     override {
     // TODO: add caching support for models.
-    if (Model)
+    if (Model) {
```

Comment:
this is the memory caching layer, I don't think alive code wants to go here

Reply:
This is the solver call invoked by souper-check though.
Hooking up alive here invokes it fine.
What am I missing?

Reply:
if it's connected with alive it goes in the base isValid call, not in the caching layers

Reply:
ok there are three isValid() implementations in this file, right?
the top one is where stuff happens, that's the one you need to modify
the other two are in the caching layers, they absolutely do not talk to alive
not sure what else I can say here...

Reply:
it's irrelevant if this code works, it's completely in the wrong place